docs/source/app-dev/app-arch.rst

```diff
@@ -1,295 +1,121 @@
+corresponding typescript data definitions for the data types declared in DAML.
 
-For all except the simplest applications, we generally recommend the state-driven approach. State-driven applications are easier to reason about when determining correctness, so this makes design and implementation easier.
+.. code-block:: bash
 
-In practice, most applications are actually a mixture of the two styles, with one predominating. It is easier to add some event handling to a state-driven application, so it is better to start with that style.
+  daml codegen ts .daml/dist/<your-project-name.dar> -o daml-ts
 
-Structuring an application
-**************************
+This will create a typescript project under ``daml-ts`` that you can install by changing directory
+to ``daml-ts`` and running ``yarn install`` and then in the root of your project ``yarn workspaces
+run build``. To start the DAML sandbox, run the ``daml-start.sh`` script at the root of the project
+directory of ``create-daml-app``.
 
-Although applications that communicate with the ledger have many purposes, they generally have some common features, usually related to their style: event-driven or state-driven. This section describes these commonalities, and the major functions of each of these styles.
+If you choose a different Javascript based frontend framework, the packages ``@daml/ledger``,
+``@daml/types`` and the generated ``@daml2ts`` package provide you with the necessary interface code
+to connect and issue commands against your ledger.
 
-In particular, all applications need to handle the asynchronous nature of the ledger API. The most important consequence of this is that applications must be multi-threaded. This is because of the asynchronous, separate streams of commands, transaction and completion events.
+Backend
+~~~~~~~
 
-Although you can choose to do this in several ways, from bare threads (such as a Java Thread) through thread libraries, generally the most effective way of handling this is by adopting a reactive architecture, often using a library such as `RxJava <https://github.com/ReactiveX/RxJava>`__.
+The backend for your application can be any DAML ledger implementation. For applications where
+performance is not of outmost importance, we recommend using the :ref:`DAML JSON API <json-api>` as
+an interface. It is served by the HTTP JSON API server connected to the ledger API server. It
+provides simple HTTP endpoints to interact with the ledger via GET/POST requests. If performance
+does become a concern for you application, you can drop the HTTP JSON API server and use the
+:ref:`gRPC API <grpc>` directly.
 
-All the language bindings support this reactive pattern as a fundamental requirement.
+When you use the ``create-daml-app`` template application you can start a local sandbox together
+with a JSON API server by running
 
-.. _event-driven-applications-1:
+.. code-block:: bash
 
-Structuring event-driven applications
```

Comment:
I think the docs on event-driven applications and state-driven applications are still relevant and useful. They are mostly independent of your client stack (unless your client stack forces you into one). E.g., I’ve had discussions around whether triggers should be state or event-based fairly frequently. Not quite sure where to integrate it but it would be a shame to drop it completely.

Reply:
I'm still quite undecided about the overall structure of the text, I think this part may survive, but maybe closer somewhere to the triggers?

Reply:
The trigger API pretty much forces you to write state-based automation. It probably makes more sense to have something a bit more generic that talks about ledger clients in general.

Reply:
My 2 cents are that this page should be just about the recommended architecture, straightforward and simple. Stat vs. event driven application discussion should be placed somewhere else (with concrete examples, current text in the docs reads like theory more).

Reply:
I also think this part needs to go somewhere else. It's way too scary for someone who just gets started.

Reply:
I fully agree! I just would like for it to not be completely lost.

Reply:
I'll move it to a .back file for now and we figure out what to do with it later. Ok?

Reply:
:+1: